packages/libs/client/src/signing/keypair/tests/signWithKeypair.test.ts

```diff
@@ -0,0 +1,230 @@
+import type { ICoin } from '../../../composePactCommand/test/coin-contract';
+import type { ISignFunction, ISingleSignFunction } from '../../../index';
+import { Pact } from '../../../index';
+import { getModule } from '../../../pact';
+import type { IKeypair } from '../createSignWithKeypair';
+import { createSignWithKeypair } from '../createSignWithKeypair';
+
+const keyPair: IKeypair = {
+  publicKey: '09e82da78d531e2d16852a923e9fe0f80f3b67a9b8d92c7f05e4782222252e12',
+  secretKey: '76e1cabaa58a33321982e434f355dc7a4cfbee092a4ac1c7aac26302ba80d992',
+};
+
+const coin: ICoin = getModule('coin');
+
+describe('createSignWithKeypair', () => {
+  let signWithKeypair: ISignFunction & ISingleSignFunction;
```

Comment:
`ISignFunction` is enough.
I think you only need to use `ISignFunction` in other places in this file as well instead of (`ISingleSignFunction` or `ISignFunction & ISingleSignFunction` ) because createSignWithKeypair always returns a function that accepts both.

Reply:
Fixed